exonum/src/events/noise/mod.rs

```diff
@@ -42,7 +43,8 @@ pub const MAX_MESSAGE_LENGTH: usize = 65_535;
 pub const TAG_LENGTH: usize = 16;
 pub const HEADER_LENGTH: usize = 4;
 
-type HandshakeResult<S> = Box<dyn Future<Item = Framed<S, MessagesCodec>, Error = io::Error>>;
+type HandshakeResult<S> =
+    Box<dyn Future<Item = (Framed<S, MessagesCodec>, x25519::PublicKey), Error = io::Error>>;
```

Comment:
I don't like this change; it leaks implementation details about the handshake (e.g., this makes it impossible to create a no-op implementation, or an implementation based on other ciphersuites). OTOH, I'm not sure about a better implementation. Could it make sense to inject the authorization logic directly into `Handshake`, i.e., make it return an error if the connecting peer is ineligible?

Reply:
I thought about it and doesn't look like a better solution, because therefore we need to maintain and update `ConnectList` on the network's side.